### PR TITLE
improve appearance

### LINKED ISSUE #235: Show tags and categories within notebook posts
Currently tags and categories are seen when searching for a notebook (i.e. from [tag pages](https://pymc-examples.readthedocs.io/en/latest/blog/tag/pymc3model.html)), but not once the notebook is [open](https://pymc-examples.readthedocs.io/en/latest/time_series/MvGaussianRandomWalk_demo.html). Might be natively supported by [ablog](https://ablog.readthedocs.io/en/latest/)

### PR DESCRIPTION
**Pull request by OriolAbril**
Work on the overall appearance of the website.

* Add "view on github", "launch on binder" and "open with collab" badges to all **posts**. Only
  notebooks with the post directive will have this. Partial resolution of https://github.com/pymc-devs/pymc-examples/issues/200, and it is automated.
    * binder env still needs to be configured though.
* Add postcard information to the left sidebar. This will include tags and categories (thus this fixes #235) but it could
  be interesting to include authors too as discussed in #198.
    * The current proposal is a bit clunky but might work well enough. It does have a bit of repeated info though which is not ideal. I have basically hijacked the ablog template to include the authors with a link to `#authors` which should then be set and contain the authorship info, links to the profiles of authors...
* ~Make the sidebar scrollable, see https://github.com/pydata/pydata-sphinx-theme/issues/500~ This will soon be fixed directly by the theme in a much more pretty way than what I first did here.
* Fix logos and PyMC3 instances in sphinx config; the notebooks and tags still need updating.

### CODE REVIEW

**Comment by OriolAbril** [comment]
Here is how the spline example looks like for example: https://pymc-examples--239.org.readthedocs.build/en/239/splines/spline.html which is where I have added authors to the post directive to show how it looks like cc @tjburch

**Comment by martinacantaro** [comment]
This is not a hard preference, but I would prefer authorship to be at the end of the notebook, and in line with [this proposal](https://github.com/pymc-devs/pymc-examples/issues/198#issuecomment-916486451). I like it better because it has more nuance than displaying the names of everybody who made a contribution, which would put someone who fixed a typo at the same level of someone who created the entire thing. 

I think this is useful for readers in cases where they want to know who did the main work for a notebook (for example, if they need to quote the notebook, or if they want to get in touch with the author). 

Also, I think for the purposes of documentation it's not as important to display authorship prominently, so I would leave the sidebar for categories and tags.

**Comment by OriolAbril** [comment]
todos: 

* [ ] update the requirements to codeautolink released version, not github
* [ ] add doi link and code to html_config once we have them. There is an issue with the webhook right now and they have not been generated even after my fake releases.
* [ ] (for someone else who knows pre-commit to do) Add a pre-commit check to ensure all notebooks end with the include page_footer directive and if possible add it automatically when missing.

**Comment by OriolAbril** [comment]
Now realized a bit of info on codeautolink is probably in order. It is a sphinx extension that automatically converts the python objects from the libraries configured on intersphinx-mappings to links to their relevant api pages. It is very new and still needs some tweaks, but it is active and mostly works.

Here are some examples form their docs: https://sphinx-codeautolink.readthedocs.io/en/latest/examples.html

**Comment by OriolAbril** [comment]
Ready to merge and I'll merge this weekend even without any other approving review.

**Review by martinacantaro — APPROVED** [review]

File: examples/splines/spline.ipynb [hunk 1]
```diff
@@ -4,16 +4,12 @@
    "cell_type": "markdown",
```
Comment (michaelosthege) [inline examples/splines/spline.ipynb:4]: <pre class="ql-syntax" spellcheck="false">that are tied togehter at their boundaries
                  ☝
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/239/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by michaelosthege — APPROVED** [review]
Found a typo that was there before.

Also the spline notebook still uses "PyMC3", not sure if you want to change that already.

**Comment by OriolAbril** [comment]
> Also the spline notebook still uses "PyMC3", not sure if you want to change that already.

We'll need to rerun them all with v4, so we can take care of that then. So far I have not updated that anywhere in notebooks, only in readme and templates.

**Pull request merged**